### ISSUE TITLE
Bad rendering of macro functions without arguments

### ISSUE THREAD

**Issue by serge-sans-paille** [michaeljones/breathe#233]
a macro function definition like 

```
#define FOO() foo
```

is rendered without parenthesis, which make it look like a macro variable.

The following hacky monkey patch fixes the issue for me:

```
# patch macro rendering by breathe
import breathe.renderer.rst.doxygen.compound as compound
def patched_update_signature(self, signode):
    if len(self.data_object.param) == 1 and not self.data_object.param[0].defname:
        signode.append(self.node_factory.Text('()'))

compound.DefineMemberDefTypeSubRenderer.update_signature = patched_update_signature
```

but there should be a better way :-/

**Comment by vitaut** [comment]
Reported to Sphinx https://github.com/michaeljones/breathe/issues/242. It's probably easier (and more reliable) to fix there than to implement a workaround in Breathe.

**Comment by melvinvermeeren** [comment]
Upstream issue is sphinx-doc/sphinx#2410.

**Comment by jakobandersen** [comment]
This should now be fixed in the 1.7 branch of Sphinx.

**Comment by melvinvermeeren** [comment]
Thank you. I'll keep this open until Sphinx 1.7.6 has been released.
